sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -1253,4 +1255,131 @@ object IntervalUtils {
     }
     intervalString
   }
+
+  def byteAndShortToYearMonthInterval(v: Int, endField: Byte): Int = {
```

Comment:
I feel it's a bit overkill to add this method to avoid overflow check for byte and short, as these two types are rarely used and perf doesn't matter that much.

I think we can just call `intToYearMonthInterval` for byte and short.